tests/ui/type/type-check/point-at-inference-4.rs

```diff
@@ -0,0 +1,24 @@
+struct S<A, B>(Option<(A,B)>);
+
+impl<A, B> S<A, B> {
+    fn infer(&self, a: A, b: B) {}
+    //~^ NOTE associated function defined here
+    //~| NOTE
+    //~| NOTE
```

Comment:
I've never made one of these before so I kinda just threw stuff at it until it passed. Are these really needed?

Reply:
Yeah, I think so.